x-pack/elastic-agent/pkg/agent/control/server/listener.go

```diff
@@ -17,6 +17,9 @@ import (
 
 func createListener() (net.Listener, error) {
 	path := strings.TrimPrefix(control.Address(), "unix://")
+	if _, err := os.Stat(path); !os.IsNotExist(err) {
+		os.Remove(path)
```

Comment:
can we check for errors here, if file exists but you're revoked access to it (too many fd, process reading/writing to it already whatever...) this will fail and we wont know

Reply:
I will log it, but the error of not creating the listener socket will still be the really reported error.